schema/schema.go

```diff
@@ -0,0 +1,70 @@
+package schema
+
+import (
+	"os"
+	"strings"
+
+	"github.com/sketch-hq/gql-lint/introspection"
+	"github.com/sketch-hq/gql-lint/parser"
+	"github.com/vektah/gqlparser/v2/ast"
+	gqlvalidator "github.com/vektah/gqlparser/v2/validator"
+)
+
+const deprecated = `directive @deprecated(
+	reason: String = "No longer supported"
+) on FIELD_DEFINITION | ARGUMENT_DEFINITION | ENUM_VALUE | INPUT_FIELD_DEFINITION`
+
+func Load(source string) (*ast.Schema, error) {
+	loader := schemaSources(source)
+	sources, err := loader.Load(source)
+	if err != nil {
+		return nil, err
+	}
+
+	return parser.ParseSchema(sources...)
+}
+
+type SchemaSource interface {
+	Load(source string) ([]*ast.Source, error)
+}
+
+type FileSchemaSource struct{}
+
+func (s FileSchemaSource) Load(source string) ([]*ast.Source, error) {
```

Comment:
More a question than a comment. I remember reading that pointers were preferred as the caller, but I don't remember what was the reasoning behind it and in this case even if it makes sense

Reply:
It.... depends. If the struct is simple it is often faster to just pass by value as there's additional "overhead" for pointer lookups. In this case we don't need to modify the struct itself either. But for this case and our use case it doesn't matter of course.